integrations/tensorflow/compiler/LegalizeTF.cpp

```diff
@@ -85,6 +86,9 @@ class LegalizeTF : public PassWrapper<LegalizeTF, FunctionPass> {
     FrozenRewritePatternList frozenPatterns(std::move(patterns));
     FrozenRewritePatternList frozenCanonicalizePatterns(
         std::move(canonicalizePatterns));
+    if (failed(applyPatternsAndFoldGreedily(op, frozenCanonicalizePatterns))) {
+      return signalPassFailure();
+    }
```

Comment:
What prompted this change? Do you have a case where this is needed that you could write a test for?

Also, could you rework the loop to call this before the `break`, instead of duplicating the logic?

Reply:
I updated the E2E tests that this affects (space_to_batch and batch_to_space).  Reworked to order the canonicalization before hand. It doesn't run after anymore however that shouldn't matter (there is a dedicated pass after).